vnext/Microsoft.ReactNative/ReactHost/CrashManager.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "CrashManager.h"
+#include "ReactInstanceWin.h"
+
+#include <winrt/Windows.Storage.h>
+
+#include <WerApi.h>
+
+namespace Mso::React {
+
+// When calling SetUnhandledExceptionFilter the previous filter is returned.
+// That should be saved so that if needed it can be called after the new unhandled exception has finished executing.
+// This allows the Windows Error Reporting system to process the error and upload the data for processing.
+static LPTOP_LEVEL_EXCEPTION_FILTER g_previousExceptionFilter = nullptr;
+
+static std::wstring g_logFileName;
+static bool g_WERExceptionFilterWasCalled{false}; // Prevent recursion in the custom handler
+
+/*static*/ std::atomic_uint CrashManager::_regCount{0};
+
+extern "C" LONG WINAPI CustomWERExceptionFilter(LPEXCEPTION_POINTERS const exceptionPointers) {
+  if (g_WERExceptionFilterWasCalled) {
+    return EXCEPTION_EXECUTE_HANDLER;
+  }
+
+  g_WERExceptionFilterWasCalled = true;
+
+  CrashManager::OnUnhandledException();
+
+  return g_previousExceptionFilter(exceptionPointers);
+}
+
+void InternalRegisterCustomHandler() noexcept {
+  // Do this now because by the time we catch the exception we may be in OOM
+#ifndef CORE_ABI // win32 vs uwp file permissions
+  wchar_t currentDirectory[MAX_PATH];
+  GetTempPath(MAX_PATH, currentDirectory);
+  g_logFileName = currentDirectory;
+#else
+  g_logFileName = winrt::Windows::Storage::ApplicationData::Current().LocalFolder().Path() + L"\\";
+#endif
+
+  g_logFileName += L"ReactNativeCrashDetails.txt";
```

Comment:
if there are multiple RNW processes on the box that are crashing, couldn't we end up getting the "wrong" file? should we append the PID or something to make it unique-ish?